src/core/tsi/ssl_transport_security.c

```diff
@@ -1013,6 +1062,10 @@ static tsi_result ssl_handshaker_create_frame_protector(
 static void ssl_handshaker_destroy(tsi_handshaker *self) {
   tsi_ssl_handshaker *impl = (tsi_ssl_handshaker *)self;
   SSL_free(impl->ssl); /* The BIO objects are owned by ssl */
+  if (impl->factory_ref) {
```

Comment:
`impl->factory_ref != NULL`

minor style issue: show that we're comparing pointers.

Reply:
Actually, you don't even need this check since `_unref` can take a `NULL` pointer.

Reply:
fixed.